rust/cardano-blockchain-types/src/metadata/cip36/key_registration.rs

```diff
@@ -111,12 +114,12 @@ impl Decode<'_, ProblemReport> for Cip36KeyRegistration {
                         cip36_key_registration.stake_pk = stake_pk;
                     },
                     Cip36KeyRegistrationKeys::PaymentAddr => {
-                        let shelley_addr = decode_payment_addr(d, err_report)?;
-                        cip36_key_registration.is_payable = shelley_addr
+                        let address = decode_payment_addr(d, err_report)?;
+                        cip36_key_registration.is_payable = address
                             .as_ref()
-                            .map(|addr| !addr.payment().is_script())
+                            .map(|addr| !VALID_PAYABLE_HEADER.contains(&addr.typeid()))
```

Comment:
seems this should be 
```suggestion
                            .map(|addr| VALID_PAYABLE_HEADER.contains(&addr.typeid()))
```